### PR TITLE
Redux toolkit query

### PR DESCRIPTION
**Pull request by kokorin**

### CODE REVIEW

File: client/src/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import { AppState } from "./store";
+import { TypedUseSelectorHook, useSelector } from "react-redux";
+
+export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector;
```
Comment (aleksashyn) [inline client/src/hooks.ts:4]: unused

File: client/src/auth.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    name: "auth",
+    initialState,
+    reducers: {
+        notAuthorized: (state) => {
+            state.isAuthenticated = false;
+        },
+    },
+});
+
+export const selectAuth: (state: AppState) => Auth = (state: AppState) => state.auth;
```
Comment (aleksashyn) [inline client/src/auth.ts:22]: this is a mutable selector, better to use createSelector

File: client/src/api/api.ts [hunk 3]
```diff
@@ -1,52 +1,28 @@
...
-    }
-    //show "something happened
-};
-
-// openapi description specifies 8080 port
-// this configuration overrides port
-const controllerConfiguration = new Configuration({
-    basePath: "",
+export const api = generatedApi.enhanceEndpoints({
+    endpoints: {},
```
Comment (aleksashyn) [inline client/src/api/api.ts:7]: looks like all endpoints without caching, what about providesTags/invalidatesTags?

File: client/src/screens/AnimeTable.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { useFindAnimeQuery } from "../api";
+import { TableCell, TableRow } from "@material-ui/core";
+import { Link } from "react-router-dom";
+import { resolveAnimeTitle } from "../service";
+
+export const AnimeTable: FC = () => {
+    const [page, setPage] = useState(0);
+    const [size, setSize] = useState(10);
+    const [title, setTitle] = useState("");
+    const { data, isLoading } = useFindAnimeQuery({ page, size, title });
```
Comment (aleksashyn) [inline client/src/screens/AnimeTable.tsx:12]: if query failed then component always render `Loading...`

File: client/src/screens/AnimeTable.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const animeTableHead = (
+        <TableRow>
+            <TableCell>ID</TableCell>
+            <TableCell align="center">TYPE</TableCell>
+            <TableCell align="center">TITLE</TableCell>
+        </TableRow>
+    );
+
+    if (!data || isLoading) {
+        return <div>Loading...</div>;
```
Comment (aleksashyn) [inline client/src/screens/AnimeTable.tsx:23]: you can use <Loader />

File: client/src/hooks.ts (hunk 1 above)
Comment (kokorin) [reply client/src/hooks.ts:4]: Created an issue

Comment (kokorin) [reply client/src/hooks.ts:4]: created an issue

File: client/src/auth.ts (hunk 2 above)
Comment (kokorin) [reply client/src/auth.ts:22]: created an issue

File: client/src/api/api.ts (hunk 3 above)
Comment (kokorin) [reply client/src/api/api.ts:7]: Created an issue to provide tags.  Cache indeed works.

File: client/src/screens/AnimeTable.tsx (hunk 4 above)
Comment (kokorin) [reply client/src/screens/AnimeTable.tsx:12]: created an issue

File: client/src/screens/AnimeTable.tsx (hunk 5 above)
Comment (kokorin) [reply client/src/screens/AnimeTable.tsx:23]: `<Loader/>` ? Created an issue.

**Pull request merged**